cpp/src/arrow/util/vector.h

```diff
@@ -130,7 +130,7 @@ Result<std::vector<T>> UnwrapOrRaise(std::vector<Result<T>>&& results) {
     }
     out.push_back(it->MoveValueUnsafe());
   }
-  return out;
+  return std::move(out);
```

Comment:
Looks to me this `std::move` does no harm.
Regarding RVO, as we are returning `Result` wrapped object, not the object itself, to my knowledge, there must be a temporary object to be moved or copied by `Result`.

Reply:
cc @pitrou @emkornfield

Reply:
I agree, it's harmless and not actively used, so this one instance is not really something to worry over much about.  Thinking on it more I think I was more surprised we get away with it everywhere else.  We are returning `Result` most of the time and we don't usually move return values.

Reply:
Agreed with @cyb70289 . RVO is only guaranteed if the returned value has the exact return type declared by the function.

Reply:
@westonpace , also change line 95? And maybe tweak commit message a bit?
I do see `return std::move(local_obj)` is used extensively when returning `Result` in other source files.